File: pset6/readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

letters = sentences = 0
# to take the two words before and after space
words = 1
# check
for ch in text:
    if ch.isalpha():
        letters += 1
    if ch.isspace():
        words += 1
    if ch in ['.', '!', '?']:
        sentences += 1
# equation
L = (letters * 100.0) / words
S = (sentences * 100.0) / words
# u can add 0.5 to round it don't know why
Grade = int(round(0.0588 * L - 0.296 * S - 15.8))
# print grades
if Grade < 1:
    print("Before Grade 1")
elif Grade >= 16:
    print("Grade 16+")
else:
    print(f"Grade {Grade}")

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include<string.h>
#include<math.h>
#include<ctype.h>
int main(void)
{
    int letters = 0;
    int words = 1; //Hello word is 2 words not one
    int sentences = 0;

    string text = get_string("Text: ");

    //count the number of letters, words, and sentences in the text.

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        //if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        char ch = text[i];
        if (isalpha(ch))
        {
            letters++;
        }
        //else if (text[i] == ' ')
        else if (isspace(ch))
        {
            words++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }

    //the Coleman-Liau index is computed using the formula
    float index = 0.0588 * (letters / (float) words) * 100 - 0.296 * (sentences / (float) words) * 100 - 15.8;
    if (index > 0 && index < 16)
    {
        printf("Grade %i\n", (int) round(index));
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Before Grade 1\n");
    }
    return 0;
}